homeassistant/components/switchgrid/calendar.py

```diff
@@ -0,0 +1,82 @@
+"""Support for Switchgrid Calendar platform."""
+
+import datetime
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import SwitchgridCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the calendar platform for entity."""
+    coordinator: SwitchgridCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([SwitchgridCalendarEntity(coordinator, config_entry)])
+
+
+class SwitchgridCalendarEntity(
+    CoordinatorEntity[SwitchgridCoordinator], CalendarEntity
+):
+    """A calendar entity holding Switchgrid Electric Loadshaving events."""
+
+    entity_id = "calendar.switchgrid_events"
+    _attr_has_entity_name = True
```

Comment:
You could create a device for switchgrid service. Then you can name the calendar "events"